lib/rust/parser/tests/parse.rs

```diff
@@ -0,0 +1,236 @@
+    let ast_ty = rust_to_meta[&value.reflect_type().id];
+
+    let base = code.as_bytes().as_ptr() as usize;
+    let code: Box<str> = Box::from(code);
+    let mut value_ = &value_[..];
+    let mut to_sexp = meta::ToSexp::new(&graph);
+    let hidden_tree_fields = ["spanLeftOffsetVisible", "spanLeftOffsetCodeRepr", "spanCodeLength"];
+    let hidden_tree_fields: HashSet<_> = hidden_tree_fields.into_iter().collect();
+    to_sexp.mapper(ast_ty, move |tree| {
+        lexpr::Value::list(tree.to_vec().unwrap().into_iter().filter(|val| match val {
+            lexpr::Value::Cons(cons) => match cons.car() {
+                lexpr::Value::Symbol(symbol) => !hidden_tree_fields.contains(symbol.as_ref()),
+                _ => panic!(),
+            },
+            _ => true,
+        }))
+    });
+    let ident_token = rust_to_meta[&enso_parser::syntax::token::variant::Ident::reflect().id];
+    let operator_token = rust_to_meta[&enso_parser::syntax::token::variant::Operator::reflect().id];
+    let number_token = rust_to_meta[&enso_parser::syntax::token::variant::Number::reflect().id];
+    let token_to_str = move |token: lexpr::Value| {
+        let code_repr = fields(&token).find(|(name, _)| *name == "codeRepr").unwrap().1;
+        let mut begin = None;
+        let mut len = None;
+        for (name, value) in fields(code_repr) {
+            match name {
+                "begin" => begin = Some(value.as_u64().unwrap() as u32),
+                "len" => len = Some(value.as_u64().unwrap() as u32),
+                _ => (),
+            }
+        }
+        let begin = begin.unwrap();
+        let begin = (begin as u64) | (base as u64 & !0xFFFF_FFFF);
+        let begin = if begin < (base as u64) { begin + 0x1_0000_0000 } else { begin };
+        let begin = begin as usize - base;
+        let len = len.unwrap() as usize;
+        code[begin..(begin + len)].to_owned().into_boxed_str()
+    };
+    let token_to_str_ = token_to_str.clone();
+    to_sexp.mapper(ident_token, move |token| lexpr::Value::symbol(token_to_str_(token)));
+    let token_to_str_ = token_to_str.clone();
+    to_sexp.mapper(operator_token, move |token| lexpr::Value::string(token_to_str_(token)));
+    let token_to_str_ = token_to_str;
+    to_sexp.mapper(number_token, move |token| {
+        lexpr::Value::Number(token_to_str_(token).parse::<u64>().unwrap().into())
+    });
+
+    let sexp = to_sexp.value(ast_ty, &mut value_);
+    let sexp = tuplify(sexp);
+    assert_eq!(value_, &[0; 0], "{}", sexp);
+    sexp
+}
+
+fn fields(value: &'_ lexpr::Value) -> impl Iterator<Item = (&'_ str, &'_ lexpr::Value)> {
+    value.list_iter().unwrap().filter_map(|value| match value {
+        lexpr::Value::Cons(cons) => match cons.car() {
+            lexpr::Value::Symbol(symbol) => Some((symbol.as_ref(), cons.cdr())),
+            _ => None,
+        },
+        _ => None,
+    })
+}
+
+fn tuplify(value: lexpr::Value) -> lexpr::Value {
```

Comment:
no docs (and the name is not clear enough, so even if its private, this needs docs)